client/apps/edit/components/layout/index.coffee

```diff
@@ -112,7 +129,11 @@ module.exports = class EditLayout extends Backbone.View
     @openTab $(e.target).index()
 
   onKeyup: =>
```

Comment:
Seems like you could consolidate this and `addRemoveReset`, maybe into a method called `autosave`.